packages-web/datagrid/src/Datagrid.tsx

```diff
@@ -6,10 +6,11 @@ import { Table } from "./components/Table";
 
 export default function Datagrid(props: DatagridContainerProps): ReactElement {
     const isServerSide = !(props.columnsFilterable || props.columnsSortable);
+    const isInfiniteLoad = !props.pagingEnabled && isServerSide;
+    const currentPage = isInfiniteLoad
```

Comment:
why not repeat this computation inside `setPage`? Then you don't need to recreate that callback that often + it will look more like a hook

Reply:
Should I reuse variables or repeat code?

Reply:
or extract a function

Reply:
They are being used in multiple places.. currentPage is not being used in line 27, also on 52. Same for isInfiniteLoad is being used in 10 and 28 and isServerSide is in 9 and 15.. So even extracted it I would use in more places and the final result will be the same.. I can wrap current page with `useMemo`

Reply:
I've applied memo

Reply:
No, please don't :( Let's actually leave the code how it was, now i think it is fine

Reply:
Reverted